File: 1051. Height Checker/heightChecker.js
/**
 * @param {number[]} heights
 * @return {number}
 */
var heightChecker = function(heights) {
    var expected = [...heights]
    expected.sort((a,b) => a-b)
    not_match = 0
    for (let i=0;i<=heights.length;i++) {
        if (expected[i]!==heights[i]) {
            not_match += 1
        }
    }
    return not_match
};

// Runtime 104 ms / Memory 42.3 MB
// debug: https://jsfiddle.net/

File: 1051. Height Checker/heightChecker.cpp
class Solution {
public:
    int heightChecker(vector<int>& heights) {
        // create spec
        vector<int> sort_heights = heights;
        sort(sort_heights.begin(),sort_heights.end());
        // compare spec
        int not_match = 0;
        for (int i=0; i<heights.size(); i++) {
            if (sort_heights[i]!=heights[i]) { not_match += 1 ; }
        }
        return not_match;
    }
};